### PR TITLE
test(detail/dialog): apply new url approach

### PR DESCRIPTION
**Pull request by DlgSHi**
### Proposed behaviour

Apply new `url` approach in tests for:
`detail` (timing: was `00:20` -> `00:07`),
`dialog` (timing: was  `00:46` -> `00:25`),
`dialog Actions` (timing  `00:42`).

### Current behaviour

Currently we are passing all parameters via `knobs`.

### Checklist

- [ ] Screenshots are included in the PR
- [ ] Carbon implementation and Design System documentation are congruent
- [ ] All themes are supported
- [x] Commits follow our style guide
- [ ] Unit tests added or updated
- [x] Cypress automation tests added or updated
- [ ] Storybook added or updated
- [ ] Typescript `d.ts` file added or updated

### CODE REVIEW

File: cypress/features/regression/dialog.feature [hunk 1]
```diff
@@ -1,135 +1,69 @@
...
-  Scenario: CloseIcon has the border outline
-    When closeIcon is focused
-    Then closeIcon has the border outline color "rgb(255, 181, 0)" and width "3px"
+  I want to check Dialog component properties
 
   @positive
   Scenario Outline: Set height for Dialog to <height>
-    When I set height to "<height>"
+    When I open default "Dialog" component in noIFrame with "dialog" json from "commonComponents" using "<nameOfObject>" object name
     Then Dialog height is set to "<height>"
```
Comment (KatarzynaQA) [inline cypress/features/regression/dialog.feature:7]: should be `int` not a `string`

File: cypress/features/regression/dialog.feature [hunk 2]
```diff
@@ -1,135 +1,69 @@
...
-      | extra-large  | 1080             |
-
-  @positive
-  Scenario: ShowCloseIcon can close Dialog
-    Given I uncheck showCloseIcon checkbox
-    When I check showCloseIcon checkbox
-    Then closeIcon is visible
-      And I click closeIcon
-      And Dialog is not visible
+      | sizeName     | sizePropertyInPx | nameOfObject |
```
Comment (KatarzynaQA) [inline cypress/features/regression/dialog.feature:47]: need to be formatted

File: cypress/features/regression/dialog.feature [hunk 3]
```diff
@@ -1,135 +1,69 @@
...
   Scenario Outline: Set out of scope characters to <height>
-    When I set height to <height> word
+    When I open default "Dialog" component in noIFrame with "dialog" json from "commonComponents" using "<nameOfObject>" object name
     Then Dialog height is not set to <height>
     Examples:
-      | height                       |
-      | -1                           |
-      | -10                          |
-      | !@#$%^*()_+-=~[];:.,?{}&"'<> |
+      | height                       | nameOfObject |
```
Comment (KatarzynaQA) [inline cypress/features/regression/dialog.feature:19]: all examples need to be formatted

File: cypress/features/regression/dialog.feature [hunk 4]
```diff
@@ -1,135 +1,69 @@
...
+      | -10                          | height-10 |
+      | !@#$%^*()_+-=~[];:.,?{}&"'<> | heightSpecialCharacter |
 
   @positive
   Scenario Outline: Change Dialog component title to <title>
-    When I set title to <title> word
-    Then component title on preview is <title>
+    When I open default "Dialog" component in noIFrame with "dialog" json from "commonComponents" using "<nameOfObject>" object name
+    Then component title on preview is <title> in NoIFrame
     Examples:
```
Comment (KatarzynaQA) [inline cypress/features/regression/dialog.feature:28]: all examples need to be formatted

File: cypress/features/regression/dialog.feature [hunk 5]
```diff
@@ -1,135 +1,69 @@
...
+      | mp150ú¿¡üßä                  | titleOtherLanguage |
+      | !@#$%^*()_+-=~[];:.,?{}&"'<> | titleSpecialCharacter |
 
   @positive
   Scenario Outline: Change Dialog subtitle to <subtitle>
-    When I set subtitle to <subtitle> word
-    Then component subtitle on preview is <subtitle> in IFrame
+    When I open default "Dialog" component in noIFrame with "dialog" json from "commonComponents" using "<nameOfObject>" object name
+    Then component subtitle on preview is <subtitle>
     Examples:
```
Comment (KatarzynaQA) [inline cypress/features/regression/dialog.feature:37]: all examples need to be formatted

File: cypress/features/regression/dialogActions.feature [hunk 6]
```diff
@@ -0,0 +1,62 @@
+Feature: Dialog component actions in IFrame
+  I want to check Dialog component - actions in IFrame
```
Comment (KatarzynaQA) [inline cypress/features/regression/dialogActions.feature:2]:
```suggestion
  I want to test Dialog component - actions in IFrame
```

File: cypress/support/step-definitions/common-steps.js [hunk 7]
```diff
@@ -342,6 +342,10 @@ Then('closeIcon is not visible', () => {
   closeIconButton().should('not.exist');
 });
 
+Then('closeIcon is not visible in NoIFrame', () => {
+  closeIconButtonNoIFrame().should('not.exist');
```
Comment (KatarzynaQA) [inline cypress/support/step-definitions/common-steps.js:346]: Can you change the name to `closeIconButton` and if there is a `closeIconButton` for Iframe then change it as `closeIconButtonIframe`, as we decided
```suggestion
  closeIconButton().should('not.exist');
```

File: cypress/support/step-definitions/detail-steps.js [hunk 8]
```diff
@@ -1,5 +1,10 @@
-import { childrenPreview, footnotePreview, iconSelect } from '../../locators/detail';
-import { icon } from '../../locators';
+import {
+  childrenPreview,
+  footnotePreview
+} from '../../locators/detail';
+import {
+  iconNoIframe
```
Comment (KatarzynaQA) [inline cypress/support/step-definitions/detail-steps.js:6]: please, change the name as we discussed `icon` for `noIframe` and `iconIframe` for `iFrame`

File: cypress/support/step-definitions/dialog-steps.js [hunk 9]
```diff
@@ -1,11 +1,27 @@
-import { alertDialogPreview as dialogPreview, dialogStickyFormFooter, dialogStickyFormFooterButton } from '../../locators/dialog/index';
-import { closeIconButton, backgroundUILocator, storyRoot } from '../../locators/index';
-import { positionOfElement } from '../helper';
+import { 
+  alertDialogPreviewNoIFrame as dialogPreview,
```
Comment (KatarzynaQA) [inline cypress/support/step-definitions/dialog-steps.js:2]: Why you can use only one a proper locator ?

Comment (DlgSHi) [reply cypress/support/step-definitions/dialog-steps.js:2]: to make alert locator used in dialog and as dialog :)

File: cypress/features/regression/dialog.feature [hunk 10]
```diff
@@ -66,4 +66,15 @@ Feature: Dialog component
   @positive
   Scenario: Disable background UI
     When I open default "Dialog" component in noIFrame with "dialog" json from "commonComponents" using "enableBackgroundUIFalse" object name
-    Then Background UI is disabled in NoIFrame
\ No newline at end of file
+    Then Background UI is disabled in NoIFrame
+
+  # Sticky form footer tests
```
Comment (KatarzynaQA) [inline cypress/features/regression/dialog.feature:71]: do we need this comment ?

Comment (DlgSHi) [reply cypress/features/regression/dialog.feature:71]: yes, we do, because I've moved those tests from other feature file

**Pull request merged**